modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java

```diff
@@ -0,0 +1,385 @@
+            assertTrue(isMethodCallsLogLine(logLine));
+        }
+        log.info("Disabling Method-calls component correlation logs");
+        configureCorrelationLoggingComponent(new String[] { "method-calls" }, false);
+        Thread.sleep(1000);
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isMethodCallsLogLine(logLine));
+        }
+
+        //test Synapse
+        log.info("Enabling Synapse component correlation logs");
+        configureCorrelationLoggingComponent(new String[] { "synapse" }, true);
+        InvokeTestAPI();
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isSynapseLogLine(logLine));
+        }
+        log.info("Disabling Synapse component correlation logs");
+        configureCorrelationLoggingComponent(new String[] { "synapse" }, false);
+        Thread.sleep(1000);
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isSynapseLogLine(logLine));
+        }
+    }
+
+    @Test(groups = {"wso2.am" }, description = "Testing persisted correlation component configurations ",
+            dependsOnMethods = { "testSpecificCorrelationLoggingConfigsTest" })
+    public void testPersistedCorrelationConfigs() throws Exception {
+        log.info("Enabling http, method-calls correlation component logs before a restart");
+        configureCorrelationLoggingComponent(new String[] { "http", "method-calls" }, true);
+        HTTPSClientUtils.doGet(getStoreURLHttps() + CORRELATION_CONFIG_PATH, header);
+        InvokeTestAPI();
+        String logLine;
+        resetAllLogs();
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+        }
+
+        serverConfigurationManager.restartGracefully();
+        Thread.sleep(10000);
+
+        while ((logLine = bufferedReader.readLine()) != null) {
+            if (logLine.contains("Started log handler")) {
+                break;
+            }
+        }
+
+        resetAllLogs();
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+        }
+        assertTrue(httpLog && methodCallsLog);
+
+        configureCorrelationLoggingComponent(new String[] { "http", "method-calls" }, false);
+        while ((logLine = bufferedReader.readLine()) != null) {
+            assertTrue(isHTTPLogLine(logLine) || isMethodCallsLogLine(logLine));
+        }
+    }
+
+    private void InvokeTestAPI() throws Exception {
+        HttpClient client = HttpClientBuilder.create().setHostnameVerifier(new AllowAllHostnameVerifier()).build();
+        HttpGet request = new HttpGet(getAPIInvocationURLHttp(API_CONTEXT, API_VERSION));
+        request.setHeader("Authorization", "Bearer " + accessToken);
+        request.setHeader("activityid", "9e3ec6ed-2a37-4b20-8dd4-d5fbc754a7d9");
```

Comment:
Use the constant `CORRELATION_ID` here as well

Reply:
Sure